src/commands.rs

```diff
@@ -2882,6 +2882,31 @@ don't make any changes, then the operation will be aborted.",
     Ok(())
 }
 
+fn _description_template_for_cmd_split(
+    workspace_command: &WorkspaceCommandHelper,
+    intro: &str,
+    overall_commit_description: &str,
+    diff_iter: TreeDiffIterator,
+) -> Result<String, CommandError> {
+    let result1 = format!(
+        "JJ: {intro}\n{}\nJJ: This part contains the following changes:\n",
+        overall_commit_description,
+    );
+
+    let mut formatter = PlainTextFormatter::new(vec![]);
```

Comment:
If you leave the buffer in a local variable and pass in a writer to that, you don't need to reach inside (undo the change the `formatter.rs`). Like this:

```
let mut diff_summary_bytes = vec![];
let mut formatter = PlainTextFormatter::new(&mut buf);
show_diff_summary(&mut formatter, workspace_command, diff_iter)?;
let diff_summary = String::from_utf8(diff_summary_bytes).unwrap();
```

Reply:
This works. I didn't expect `PlainTextFormatter<&mut Vec<u8>>` to be a valid type, but it seems to be.